examples/ping-pong/src/client.rs

```diff
@@ -0,0 +1,82 @@
+use crate::messages::{Ping, Pong, PING_MSG_TYPE, PONG_MSG_TYPE};
+use codec_sv2::{Frame, StandardDecoder, StandardSv2Frame};
+use std::{
+    io::{Read, Write},
+    net::TcpStream,
+};
+
+use crate::error::Error;
+
+pub fn start_client(address: &str) -> Result<(), Error> {
+    let mut stream = TcpStream::connect(address)?;
+
+    println!("CLIENT: Connected to server on {}", address);
+
+    // create Ping message
+    let ping_message = Ping::new()?;
+    let ping_nonce = ping_message.get_nonce();
+
+    // create Ping frame
+    let ping_frame =
+        StandardSv2Frame::<Ping>::from_message(ping_message.clone(), PING_MSG_TYPE, 0, false)
+            .ok_or(Error::FrameFromMessage)?;
+
+    // encode Ping frame
+    let mut encoder = codec_sv2::Encoder::<Ping>::new();
+    let ping_encoded = encoder.encode(ping_frame)?;
+
+    println!("CLIENT: Sending Ping to server with nonce: {}", ping_nonce);
+    stream.write_all(ping_encoded)?;
+
+    // ok, we have successfully sent the ping message
+    // now it's time to receive and verify the pong response
+
+    // initialize decoder
+    let mut decoder = StandardDecoder::<Ping>::new();
+
+    // right now, the decoder buffer can only read a frame header
```

Comment:
sorry, I am not getting why the first read fails, could you pleas elaborate?

Reply:
> It seems like we're only reading the first 6 bytes of this specific message type and not reaching the total byte length, which causes an error signaling that more bytes are needed to match the payload size. 

correct

> But I'm wondering, why are we taking this approach? Couldn't we just read all the bytes in one go and then decipher from there?

this is technical debt we need to clean up

https://github.com/stratum-mining/stratum/issues/903 is the first step of that journey